test/integration/kubernetes.go

```diff
@@ -72,6 +84,9 @@ func (ke *k8sEnvironment) addPod(t *testing.T, pod *corev1.Pod) *corev1.Pod {
 	p, err := ke.client.CoreV1().Pods(ke.testNamespace.Name).Create(context.Background(), pod, metav1.CreateOptions{})
 	require.NoError(t, err)
 
+	// Add a small delay to ensure that pod is fully created.
+	time.Sleep(3 * time.Second)
```

Comment:
Pod creation can be very slow, 3 seconds is not going to make it. It would make more sense to check for the pod to be ready [here is an example script that checks pods for a specific condition](https://github.com/newrelic/nri-kubernetes/blob/main/internal/testutil/datagen/datagen.sh#L246-L259) and an [example usage of that script](https://github.com/newrelic/nri-kubernetes/blob/main/internal/testutil/datagen/datagen.sh#L216-L217)

Reply:
I guess this comment was rather poorly written in retrospect. I just added a delay for the pod creation request to "finish sending". The `addPodAndWaitOnPhase()` which calls `addPod()` already has a try and await cycle to ensure the pod is active. Without the wait, I've seen that the wait loop occasionally fails. Thoughts?

Reply:
In that case, I would suggest you wait here for the [Pending state](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase) by calling `addPodAndWaitOnPhase`. That will make tests more robust to delays in accepting the request